activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/main/java/org/activiti/cloud/services/query/events/handlers/BPMNSequenceFlowTakenEventHandler.java

```diff
@@ -38,9 +38,32 @@ public BPMNSequenceFlowTakenEventHandler(EntityManager entityManager) {
 
     @Override
     public void handle(CloudRuntimeEvent<?, ?> event) {
-        CloudSequenceFlowTakenEvent activityEvent = CloudSequenceFlowTakenEvent.class.cast(event);
+        CloudSequenceFlowTakenEvent sequenceFlowTakenEvent = CloudSequenceFlowTakenEvent.class.cast(event);
+        var entityId = getEntityId(sequenceFlowTakenEvent);
 
-        BPMNSequenceFlow bpmnSequenceFlow = activityEvent.getEntity();
+        Optional
+            .ofNullable(entityManager.find(BPMNSequenceFlowEntity.class, entityId))
+            .ifPresentOrElse(
+                sequenceFlowEntity -> {
+                    logger.warn(
+                        "Sequence flow '" +
+                        sequenceFlowEntity.getElementId() +
+                        "' with eventId '" +
+                        event.getId() +
+                        "' already exists in the process '" +
+                        sequenceFlowTakenEvent.getEntity().getProcessInstanceId() +
+                        "'!"
+                    );
+                },
+                () -> {
+                    var sequenceFlowTakenEntity = createBpmnSequenceFlowEntity(entityId, sequenceFlowTakenEvent);
+                    entityManager.persist(sequenceFlowTakenEntity);
+                }
+            );
+    }
```

Comment:
Do we need to do this existence test? I think that once we are using the combination of event id, elemnt id and process id as index, we can skip the checking and use the `entityManager.persist` that is idempotent